### PR TITLE
Avner/DEV2-2244 onprem

### PR DESCRIPTION
**Pull request by avnerbarr**

### CODE REVIEW

File: .github/workflows/packageOnPrem.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.19'
+      #          runs the binaries downloader script which will be packaged into this plugin
+      - name: pull binaries
+        run: VERSION=${{ inputs.binaryVersion }} ./binaries/downloadBinaries.sh
+      #      https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
+      - name: modify readme
+        run: |
+          echo "" > ./README.md
```
Comment (assapir) [inline .github/workflows/packageOnPrem.yml:30]: Why deleting the current Readme and not just adding to it?

File: buildOnPrem.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+yarn onprem:prep \
+&& yes | vsce package \
```
Comment (assapir) [inline buildOnPrem.sh:2]: Make this file a yarn script?

File: package.json [hunk 3]
```diff
@@ -307,6 +320,10 @@
             "default": null,
             "description": "Experimental: Enable modules auto-import following applying Tabnine suggestions"
           },
+          "tabnine.cloudHost": {
+            "type": "string",
+            "description": "The tabnine cloud host to connect to"
```
Comment (assapir) [inline package.json:325]: Can we expose that only if on prem?

File: src/binary/runBinary.ts [hunk 4]
```diff
@@ -13,14 +14,17 @@ export default async function runBinary(
...
   const context = getTabnineExtensionContext();
   const args: string[] = [
-    "--client=vscode",
+    ON_PREM ? "--client=vscode_on_prem" : "--client=vscode",
     "--no-lsp=true",
     tabnineExtensionProperties.logFilePath
       ? `--log-file-path=${tabnineExtensionProperties.logFilePath}`
       : null,
+    ON_PREM ? "--onprem=true" : null,
+    ON_PREM ? "--no_bootstrap" : null,
```
Comment (assapir) [inline src/binary/runBinary.ts:23]: No need to pass it, it won't bootstrap anyway

File: src/binary/runBinary.ts [hunk 5]
```diff
@@ -13,14 +14,17 @@ export default async function runBinary(
...
     tabnineExtensionProperties.logLevel
       ? `--log-level=${tabnineExtensionProperties.logLevel}`
       : null,
+    tabnineExtensionProperties.cloudHost ? `--cloud2_url=${tabnineExtensionProperties.cloudHost}` : null,
```
Comment (assapir) [inline src/binary/runBinary.ts:27]: Pass --onprem only if that set?

File: src/onPrem.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+var PREM = require('./onPremJ.js');
```
Comment (assapir) [inline src/onPrem.ts:1]: Wait what?
Also, use const, not var

File: .github/workflows/packageOnPrem.yml (hunk 1 above)
Comment (avnerbarr) [reply .github/workflows/packageOnPrem.yml:30]: because it has lots of links and junk

File: buildOnPrem.sh (hunk 2 above)
Comment (avnerbarr) [reply buildOnPrem.sh:2]: can't because it is modifying the package.json and that didn't work

File: package.json (hunk 3 above)
Comment (avnerbarr) [reply package.json:325]: yes it can be possible to add in the prep phase

File: src/binary/runBinary.ts (hunk 4 above)
Comment (avnerbarr) [reply src/binary/runBinary.ts]: I'm not sure about that, depends on the build

File: src/binary/runBinary.ts (hunk 5 above)
Comment (avnerbarr) [reply src/binary/runBinary.ts]: the other way around otherwise it won't work in regular mode. fixed that

File: src/onPrem.ts (hunk 6 above)
Comment (avnerbarr) [reply src/onPrem.ts]: will that work with a require call?

File: package.json (hunk 3 above)
Comment (avnerbarr) [reply package.json:325]: any idea how to append a nested object to this json from a cli call? (yarn,npm etc.) (without installing some fancy tool I suppose)

Comment (avnerbarr) [reply package.json:325]: I suppose this might work. I'll try 
https://www.npmjs.com/package/transform-json-webpack-plugin

**Comment by avnerbarr** [comment]
moved to this PR because of JIRA rule
https://github.com/codota/tabnine-vscode/pull/1033

**Pull request closed without merging**